hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestColumnStatsIndex.scala

```diff
@@ -457,23 +462,24 @@ class TestColumnStatsIndex extends HoodieClientTestBase {
 
   private def sort(df: DataFrame): DataFrame = {
     val sortedCols = df.columns.sorted
-    // Sort dataset by the first 2 columns (to minimize non-determinism in case multiple files have the same
+    // Sort dataset by 4 columns (to minimize non-determinism in case multiple files have the same
     // value of the first column)
     df.select(sortedCols.head, sortedCols.tail: _*)
-      .sort("c1_maxValue", "c1_minValue")
+      .sort("c1_maxValue", "c1_minValue", "c2_maxValue", "c2_minValue")
```

Comment:
The log file has the same record key range (min and max of "c1" column) as the parquet base file so we need to add two more columns for deterministic ordering.